L/Leptonica/build_tarballs.jl

```diff
@@ -27,7 +28,7 @@ platforms = supported_platforms()
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("liblept", :liblept),
+    LibraryProduct("libleptonica", :liblept),
```

Comment:
The change of the soname will actively break dependents.  It'd be good to fix compat bounds of dependents jll before moving forward with this PR.